nibabel/tests/test_recoder.py

```diff
@@ -12,50 +12,75 @@
 
 from ..volumeutils import Recoder, DtypeMapper, native_code, swapped_code
 
-from nose.tools import assert_equal, assert_raises, assert_true, assert_false
+import pytest
 
 
-def test_recoder():
+def test_recoder_1():
     # simplest case, no aliases
     codes = ((1,), (2,))
     rc = Recoder(codes)
-    yield assert_equal, rc.code[1], 1
-    yield assert_equal, rc.code[2], 2
-    yield assert_raises, KeyError, rc.code.__getitem__, 3
+    assert rc.code[1] == 1
+    assert rc.code[2] == 2
+    with pytest.raises(KeyError):
+        rc.code.__getitem__(3)
+
+def test_recoder_2():
     # with explicit name for code
+    codes = ((1,), (2,))
     rc = Recoder(codes, ['code1'])
-    yield assert_raises, AttributeError, rc.__getattribute__, 'code'
-    yield assert_equal, rc.code1[1], 1
-    yield assert_equal, rc.code1[2], 2
+    with pytest.raises(AttributeError):
+        rc.__getattribute__('code')
```

Comment:
And `__getattribute__` and `getattr`:

```suggestion
        rc.code
```